packages/app/src/systems/Settings/hooks/useSettings.ts

```diff
@@ -0,0 +1,65 @@
+import { useInterpret, useSelector } from '@xstate/react';
+import { useNavigate } from 'react-router-dom';
+
+import type { SettingsMachineState } from '../machines/settingsMachine';
+import { settingsMachine } from '../machines/settingsMachine';
+
+import type { AccountInputs } from '~/systems/Account';
+import { useAccount } from '~/systems/Account';
+import { Pages } from '~/systems/Core';
+
+const selectors = {
+  isChangingPassword: (state: SettingsMachineState) =>
+    state.matches('changingPassword'),
+
+  hasChangedPassword: (state: SettingsMachineState) => state.matches('done'),
+
+  isUnlocking: (state: SettingsMachineState) =>
+    state.matches('unlocking') || state.matches('gettingMnemonic'),
+
+  words: (state: SettingsMachineState) => state.context.words,
+};
+
+export function useSettings() {
+  const navigate = useNavigate();
+  const { account } = useAccount();
+  const service = useInterpret(() =>
+    settingsMachine.withConfig({
+      actions: {
+        goToWallet() {
+          navigate(Pages.wallet());
+        },
+      },
+    })
+  );
+  const { send } = service;
+
+  const isUnlocking = useSelector(service, selectors.isUnlocking);
+
```

Comment:
I don't think you need to give a space between each variable, we're not following this pattern alongside the codebase